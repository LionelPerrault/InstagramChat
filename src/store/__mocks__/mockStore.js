import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
