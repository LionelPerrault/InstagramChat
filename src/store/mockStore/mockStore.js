import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'
import Post from './mock-reducers/Post'
import Hashtag from './mock-reducers/Hashtag'
import Setting from './mock-reducers/Setting'
import Message from './mock-reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message
}

const middlewares = [
  promise(), thunk
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
