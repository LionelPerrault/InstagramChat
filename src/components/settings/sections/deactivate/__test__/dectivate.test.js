import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../store/__mocks__/mockStore'
import Deactivate from '../deactivate'

describe('Deactivate Component', () => {
  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <Deactivate />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
