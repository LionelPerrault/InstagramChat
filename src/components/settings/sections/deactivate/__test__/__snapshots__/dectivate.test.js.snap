// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deactivate Component should match snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "animation": "bcCCNc 300ms ease 0s 1 normal none running",
        "backfaceVisibility": "visible",
      }
    }
  >
    <div
      className="dlt_acc"
    >
      <div
        className="c_p_header"
      >
        <span>
          Deactivate your account
        </span>
      </div>
      <form
        className="dlt_acc_form"
        onSubmit={[Function]}
      >
        <input
          autoComplete="false"
          autoFocus={true}
          disabled={false}
          maxLength="255"
          onChange={[Function]}
          placeholder="Your password.."
          required={true}
          spellCheck="false"
          type="password"
          value=""
        />
        <input
          disabled={true}
          type="submit"
          value="Deactivate"
        />
      </form>
      <div
        className="dlt_acc_info"
      >
        <span
          className="dlt_acc_bold"
        >
          Note:
        </span>
        <span>
          All of your 
          <span
            className="dlt_acc_bold"
          >
            posts, followers, followings, recommendations, messages, groups, settings, notifications, bookmarks & other info will be permanently deleted.
          </span>
           And you won't be able to find it again.
        </span>
      </div>
    </div>
  </div>
</div>
`;
